src/equations/compressible_euler_2d.jl

```diff
@@ -1404,6 +1404,17 @@ end
     return SVector(w1, w2, w3, w4)
 end
 
+# Transformation from conservative variables u to d(p)/d(u)
+@inline function pressure(u, equations::CompressibleEulerEquations2D, derivative::True)
```

Comment:
I would hesitate labeling this function `pressure` - instead, maybe use `pressure_derivative` (the term `gradient` is somewhat occupied by parabolic terms imo and probably again misleading).

Reply:
I see your point. My problem was that I want to pass the function of the specific variable with e.g. `positivity_variables_nonlinear = [pressure]`.
For the limiting, I need the pressure itself and its derivative. Since I didn't always want to pass a tuple `[(pressure, pressure_derivative)]`. Is there another way to do this cleaner?

Reply:
Would it be better to just call this function `pressure_derivative()` and then add
```
@inline pressure(u, equations, derivative::True) = pressure_derivative(u, equations)`
```

Reply:
Ah I see, I was not aware of the dual use.
I think the latter is an elegant solution, but of course should be equipped with a comment.

Reply:
TBH, I am very skeptical of a function that is called `pressure` and that returns something that is _not_ the pressure...

Reply:
Yes, that's probably true. Can you think of another option than passing a tuple?:worried:  @sloede

Reply:
```julia
julia> function pressure end # standard Trixi.jl
pressure (generic function with 0 methods)

julia> function derivative_stuff_that_is_only_needed_for_subcell_limiting(::typeof(pressure), u, equations)
           println("pressure derivative")
       end
derivative_stuff_that_is_only_needed_for_subcell_limiting (generic function with 1 method)

julia> u, equations = 1, 2;

julia> derivative_stuff_that_is_only_needed_for_subcell_limiting(pressure, u, equations)
pressure derivative
```

Reply:
Thanks a lot. I now added such a function.

I also thought about directly using ForwardDiff.jl, but this was much slower:
```
# Hard-coded version for pressure
julia> @btime Trixi.variable_derivative($pressure, $u, $equations)
  1.753 ns (0 allocations: 0 bytes)

julia> @btime Trixi.ForwardDiff.gradient(x -> pressure(x, $equations), $u)
  12.868 ns (0 allocations: 0 bytes)
```
This was performance relevant.

I now added such an automatic derivation as a slow default option for all variables:
```
# Default implementation of derivation for `variable`. Used for subcell limiting.
# Implementing a derivative function for a specific function improves the performance.
@inline function variable_derivative(variable, u, equations)

    return ForwardDiff.gradient(x -> variable(x, equations), u)
end
```